packages/theme/src/components/skeleton.ts

```diff
@@ -1,6 +1,20 @@
-import { keyframes } from "@chakra-ui/system"
 import { getColor, mode } from "@chakra-ui/theme-tools"
 
+let emotion = null
+try {
+  // eslint-disable-next-line global-require
+  emotion = require('@emotion/react')
+} catch {
+  // eslint-disable-next-line global-require
+  emotion = require('@emotion/css')
+}
+
+if(!emotion) {
+  throw Error('Either @emotion/react (for React) or @emotion/css (for others) must be installed')
+}
+
+const { keyframes } = emotion
```

Comment:
I might need to think about this a bit more.

An alternative approach is to put the keyframe definition in the theme as an object and translate that to emotion's keyframe at the system or styled-system level.

Reply:
@segunadebayo  Or should we introduce a generalized version of it? 
Something like `CssSerializer`, some theme components may need `keyframes`, or `css`, `injectGlobal`
```ts
interface CssSerializer {
  css: (...args: any) => Dict,
  keyframes: (...args: any) => String,
  injectGlobal: (...args: any) => any
}

type StyleResolverProps = SystemStyleObject & {
  __css?: SystemStyleObject
  sx?: SystemStyleObject
  theme: any
  serializer: CssSerializer
  css?: CSSObject
}
```
And inject that serializer to `props` at the `system` level?
```ts
import emotion from "@emotion/react"
...
export const toCSSObject: GetStyleObject = ({ baseStyle }) => (props) => {
  const { theme, css: cssProp, __css, sx, ...rest } = props
  const styleProps = objectFilter(rest, (_, prop) => isStyleProp(prop))
  const finalBaseStyle = runIfFn(baseStyle, {
    serializer: emotion,
    ...props,
  })
  // ... other logics
}  
```
And use them in style function:
```ts
// skeleton.ts
const baseStyle = (props: Record<string, any>) => {
  // ...
  const {
    startColor = defaultStartColor,
    endColor = defaultEndColor,
    speed,
    theme,
    serializer,
  } = props
  // ...
  const fadeAnimation = serializer.keyframes({
    from: { borderColor: startColor, background: startColor },
    to: { borderColor: endColor, background: endColor },
  })
  
  return {
    opacity: 0.7,
    borderRadius: "2px",
    borderColor: start,
    background: end,
    animation: `${speed}s linear infinite alternate ${fadeAnimation}`,
  }
}  
```
Will this work, do I miss something?

Reply:
This looks interesting. I'll take a look at the code to see how to make this work.

Reply:
Actually, I've updated this PR to the proposed approach, you can check it now @segunadebayo